### PR TITLE
Live migration for kvm-ssh

### PR DESCRIPTION
**Pull request by xorel**

### CODE REVIEW

File: src/vmm_mad/remotes/kvm/migrate [hunk 1]
```diff
@@ -122,29 +119,47 @@ else
...
+                MIGRATE_DISKS+="${DISK_DEV},"
             fi
+
             # do initial rsync
-            exec_and_log "rsync $DISK $DEST_HOST:$DISK" \
-                "Failed to rsync disk $DISK to $DEST_HOST:$DISK"
+            multiline_exec_and_log "tar -cSf - $DISK_PATH | ssh $DEST_HOST 'tar xSf - -C / '" \
+                "Failed to rsync disk $DISK_PATH to $DEST_HOST:$DISK_PATH"
+
+        elif [ "$FORMAT" = "qcow2" ] && [ ! -L "$DISK_PATH" ]; then
```
Comment (atodorov-storpool) [inline src/vmm_mad/remotes/kvm/migrate:143]: $DISK_PATH is already tested in get_size_and_format_of_disk_img() line 45
so for the symlinked disks we will have FORMAT=network-disk and this test will be skipped on the first part so the following should be enough (IMO)?

elif [ "$FORMAT" = "qcow2" ]; then

File: src/vmm_mad/remotes/kvm/migrate [hunk 2]
```diff
@@ -122,29 +119,47 @@ else
...
+        if [ -d "${DISK_PATH}.snap" ] || [ -L "${DISK_PATH}.snap" ]; then
+            multiline_exec_and_log "tar -cSf - $DISK_PATH.snap | ssh $DEST_HOST 'tar xSf - -C / '" \
+                "Failed to rsync disk snapshot ${DISK_PATH}.snap to $DEST_HOST"
+        fi
+
+        # recreate symlinks
+        if [ -L "$DISK_PATH" ]; then
+            LINK_TARGET=$(readlink $DISK_PATH)
+            ssh_exec_and_log "$DEST_HOST" "ln -s \"$LINK_TARGET\" \"$DISK_PATH\"" \
+                "Faild to create symlink $DISK_PATH -> $LINK_TARGET on $DEST_HOST"
```
Comment (atodorov-storpool) [inline src/vmm_mad/remotes/kvm/migrate:162]: TYPO: Failed :-)

**Review by atodorov-storpool — COMMENTED** [review]
Well, this will definitely work!. Commented on some cosmetics and possible corner cases to address from the top of my head...

**Comment by xorel** [comment]
Thanks for the input, I pushed the fix.

**Pull request closed without merging**